walk-in-interview/src/main/java/com/google/job/data/JobPayment.java

```diff
@@ -5,20 +5,22 @@
     private final int min;
     private final int max;
     private final PaymentFrequency paymentFrequency;
+    private final long annualMax;
 
     private volatile int hashCode;
 
     // For serialization
     public JobPayment() {
-        this(/* min= */0, /* max= */0, PaymentFrequency.HOURLY);
+        this(/* min= */0, /* max= */0, PaymentFrequency.HOURLY, /* annualMax= */ 0);
     }
 
-    public JobPayment(int min, int max, PaymentFrequency paymentFrequency) {
+    public JobPayment(int min, int max, PaymentFrequency paymentFrequency, long annualMax) {
         validateParameter(min, max);
 
         this.min = min;
         this.max = max;
         this.paymentFrequency = paymentFrequency;
+        this.annualMax = annualMax;
```

Comment:
Currently validateParameter doesn't validate if the annualMax consistent with the previous max * frequency.